packages/flutter/lib/src/material/tabs.dart

```diff
@@ -660,6 +661,9 @@ class TabBar extends StatefulWidget implements PreferredSizeWidget {
   /// is null then the text style of the theme's body2 definition is used.
   final TextStyle unselectedLabelStyle;
 
+  /// An option callback when the [TabBar] is tapped.
```

Comment:
An optional callback that's called when the [TabBar] is tapped.

The callback is applied to the index of the tab where the tap occurred. 

This callback has no effect on the default handling of taps. Applications that want to do a little extra work when a tab is tapped, even if the tap doesn't change the TabController's index, can specify [onTap].